extension/third_party_attribute_extension.go

```diff
@@ -0,0 +1,69 @@
+package extension
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/getyoti/yoti-go-sdk/v2/attribute"
+)
+
+const (
+	thirdPartyAttributeExentionTypeConst = "THIRD_PARTY_ATTRIBUTE"
+)
+
+// ThirdPartyAttributeExtensionBuilder is used to construct a ThirdPartyAttributeExtension
+type ThirdPartyAttributeExtensionBuilder struct {
+	extension ThirdPartyAttributeExtension
+}
+
+// ThirdPartyAttributeExtension is an extension representing the issuance of a third party attribute
+type ThirdPartyAttributeExtension struct {
+	expiryDate  *time.Time
+	definitions []attribute.AttributeDefinition
+}
+
+// New initializes the builder
+func (builder *ThirdPartyAttributeExtensionBuilder) New() *ThirdPartyAttributeExtensionBuilder {
+	return builder
+}
+
+// WithExpiryDate sets the expiry date of the extension
+func (builder *ThirdPartyAttributeExtensionBuilder) WithExpiryDate(expiryDate *time.Time) *ThirdPartyAttributeExtensionBuilder {
+	builder.extension.expiryDate = expiryDate
+	return builder
+}
+
+// WithDefinition adds an attribute.AttributeDefinition to the list of definitions
+func (builder *ThirdPartyAttributeExtensionBuilder) WithDefinition(definition attribute.AttributeDefinition) *ThirdPartyAttributeExtensionBuilder {
+	builder.extension.definitions = append(builder.extension.definitions, definition)
+	return builder
+}
+
+// WithDefinitions sets the array of attribute.AttributeDefinition on the extension
+func (builder *ThirdPartyAttributeExtensionBuilder) WithDefinitions(definitions []attribute.AttributeDefinition) *ThirdPartyAttributeExtensionBuilder {
+	builder.extension.definitions = definitions
+	return builder
+}
+
+// Build creates a ThirdPartyAttributeExtension using the supplied values
+func (builder *ThirdPartyAttributeExtensionBuilder) Build() ThirdPartyAttributeExtension {
+	return builder.extension
+}
+
+// MarshalJSON returns the JSON encoding
+func (extension ThirdPartyAttributeExtension) MarshalJSON() ([]byte, error) {
+	type thirdPartyAttributeExtension struct {
+		ExpiryDate  string                          `json:"expiry_date"`
+		Definitions []attribute.AttributeDefinition `json:"definitions"`
+	}
+	return json.Marshal(&struct {
+		Type    string                       `json:"type"`
+		Content thirdPartyAttributeExtension `json:"content"`
+	}{
+		Type: thirdPartyAttributeExentionTypeConst,
```

Comment:
You can use `yotiprotoshare.DataEntry_THIRD_PARTY_ATTRIBUTE.String()` here instead of having to redeclare the constant at the top of this file
```suggestion
		Type: yotiprotoshare.DataEntry_THIRD_PARTY_ATTRIBUTE.String(),
```